static/app/views/dashboards/widgetBuilder/widgetBuilder.tsx

```diff
@@ -355,14 +355,14 @@ function WidgetBuilder({
 
   useEffect(() => {
     const onUnload = () => {
-      if (!isSubmitting && state.userHasModified) {
+      if (!isSubmittingRef.current && state.userHasModified) {
```

Comment:
in this case the state was not yet updated by the useEffect before being checked. We can use useRef because it isn't being rendered to dom.